### PR TITLE
Various fixes related to removing residual package folders.

### PR DESCRIPTION
**Pull request by YossiMH**
- Fixed https://github.com/GlitchEnzo/NuGetForUnity/issues/149 -- removing a package listing from the `packages.config` file does not remove it from the Packages folder. Added some additional unit tests to check this case.
- Changed file and folder deletion to use Unity asset database, to address some issues where asset database would recreate a folder after its contents were deleted.
- Delete any invalid package folders as the first step of doing a package restore to subsequently clean up artifacts after a deletion via the asset database failed (which can happen if you have visual studio open).

### CODE REVIEW

File: Packager/ProjectSettings/ProjectVersion.txt [hunk 1]
```diff
@@ -1,2 +1 @@
-m_EditorVersion: 5.2.0f3
```
Comment (jwittner) [inline Packager/ProjectSettings/ProjectVersion.txt]: I think we want to keep the Packager at the oldest version that NuGetForUnity intends to support.

File: .gitignore [hunk 2]
```diff
@@ -4,7 +4,7 @@
 [Bb]uild/
 
 # Visual Studio 2015 cache directory
-/.vs/
+.vs/
```
Comment (jwittner) [inline .gitignore]: Please remove this change.

File: .gitignore [hunk 3]
```diff
@@ -69,7 +69,21 @@ $RECYCLE.BIN/
 # Ignore the NuGet Packages folder
 Packages/
 Packages.meta
+/Packager/Assets/packages.config
```
Comment (jwittner) [inline .gitignore]: Please move these to a .gitignore inside the package folder.

File: .gitignore [hunk 4]
```diff
@@ -69,7 +69,21 @@ $RECYCLE.BIN/
...
+/Packager/Assets/packages.config.meta
+
+# Ignore the NuGet config file
+/Assets/NuGet.config
```
Comment (jwittner) [inline .gitignore]: This shouldn't be ignored for the NuGetForUnity project.

File: .gitignore [hunk 5]
```diff
@@ -69,7 +69,21 @@ $RECYCLE.BIN/
...
+/Assets/NuGet.config.meta
+/Packager/Assets/NuGet.config
```
Comment (jwittner) [inline .gitignore]: More to move to the Package/.gitignore file.

File: Assets/NuGet/Editor/NugetPreferences.cs [hunk 6]
```diff
@@ -11,7 +11,7 @@ public static class NugetPreferences
         /// <summary>
         /// The current version of NuGet for Unity.
         /// </summary>
-        public const string NuGetForUnityVersion = "0.0.15";
+        public const string NuGetForUnityVersion = "0.0.16-alpha.2";
```
Comment (jwittner) [inline Assets/NuGet/Editor/NugetPreferences.cs]: I don't think this should be committed.

File: Packager/ProjectSettings/ClusterInputManager.asset [hunk 7]
```diff
@@ -0,0 +1,6 @@
+%YAML 1.1
```
Comment (jwittner) [inline Packager/ProjectSettings/ClusterInputManager.asset]: Looks like you've updated the project or something and now it's adding all kinds of stuff. Can we get this change in without changing any project versions?

**Comment by YossiMH** [comment]
@jwittner -- I've addressed your change requests.

**Comment by ForrestTrepte** [comment]
Thanks for reporting this issue and submitting this issue, @YossiMH! PR #190 ended up fixing #149. I'm closing this PR for now, but if you would like to get your other changes in please merge with latest master and create a new PR.

**Pull request closed without merging**